src/api/rest/providers/github/schemas/users.js

```diff
@@ -0,0 +1,89 @@
+import { schema } from 'normalizr';
+import moment from 'moment/min/moment.min';
+
+export const userSchema = new schema.Entity(
+  'users',
+  {},
+  {
+    idAttribute: user => user.login.toLowerCase(),
+    processStrategy: entity => {
+      const processed = {};
+
+      // These are provided in both mini & full modes
+      processed.id = entity.login.toLowerCase(); // id should be always used for navigation
```

Comment:
Think we we should be able to simplify this logic with spread.

```
const processed = {
   ...entity,
   id: login.toLowerCase(),
   avatarUrl: avatar_url,
    // ....
}
```

And for keys with the same names (like `login`) we shouldn't need to have to include it as it'll be part of the spread

Reply:
We don't want to use the spread here, as you'll be copying *all* the original JSON attributes in the normalized entity.

I've willingly did an extra verbose function because we need to take really care of:

1. handpicking the attributes we will need, and only them => helps keep the store as slim as possible
2. name them carefully & clearly => because they will be used in all the app
3. construct them by hand if needed:
> imagine if Bitbucket returns { first_name: 'John', last_name: 'Doe' }, Gitlab returns { name: 'John', family_name: 'Doe' } and Github { fullname: 'John Doe' }
> we will have 3 schemas that all construct the following attribute: fullName: 'John Doe'

Of course, we can use some more syntax sugar here. For now I'm keeping things as readable as possible while this PR is being reviewed/challenged.